app/src/main/java/org/stepic/droid/ui/custom/AutoCompleteSearchView.kt

```diff
@@ -0,0 +1,84 @@
+package org.stepic.droid.ui.custom
+
+import android.app.Activity
+import android.app.SearchManager
+import android.content.Context
+import android.support.annotation.DrawableRes
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.SearchView
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import kotlinx.android.synthetic.main.search_queries_recycler_view.view.*
+import org.stepic.droid.R
+import org.stepic.droid.analytic.Analytic
+import org.stepic.droid.ui.adapters.SearchQueriesAdapter
+
+
+class AutoCompleteSearchView
+@JvmOverloads
+constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : SearchView(context, attrs, defStyleAttr) {
+    val searchQueriesAdapter = SearchQueriesAdapter()
+    var suggestionsOnTouchListener: OnTouchListener? = null
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        searchQueriesAdapter.searchView = this
+    }
+
+    override fun onDetachedFromWindow() {
+        searchQueriesAdapter.searchView = null
+        super.onDetachedFromWindow()
+    }
+
+    init {
+        maxWidth = 20000
+    }
+
+    fun initSuggestions(rootView: ViewGroup) {
+        val inflater = LayoutInflater.from(context)
+        val searchQueriesRecyclerView = inflater.inflate(R.layout.search_queries_recycler_view, rootView, false).searchQueriesRecyclerView
+        searchQueriesRecyclerView?.let {
```

Comment:
and it can't be null by design. If it is null, then layout is not found -> we have a big problem, and shouldn't ignore that